### DISCUSSION TITLE
Distorsion simulation

### DISCUSSION THREAD

**Discussion by bnilsson11** [ra3xdh/qucs_s#920] [Q&A] [answered]
Hi,

I am trying to simulate the harmonic distorsion of my circuit, starting with the distortion example.
Reading the ngspice manual, I am stuck on the following:
"To obtain HD2 and HD3, one must divide by the corresponding AC values at F1, obtained from an .ac line. This division can be done again using interactive or control commands."
How should the nutmeg equation be formulated to do this?
I am simply getting lost in the syntax...

**Comment by tomhajjar** [comment, marked as answer]
Here are some examples I have collected.

![2024-08-31_211404](https://github.com/user-attachments/assets/71d13cbd-527d-4318-921c-1d4ad1025e89)

[Dist.zip](https://github.com/user-attachments/files/16826749/Dist.zip)

**Comment by bnilsson11** [comment]
I would like to do a Parameter Sweep simulation of an AC source signal level and run TRAN and FOURIER to chart THD vs. output level of my amp.
Is this possible?
Or do I have to script it?

**Reply by ivandi69** [reply]
#1180
